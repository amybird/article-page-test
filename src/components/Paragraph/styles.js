import styled from 'styled-components';

export const Text = styled.p`
    font-family: serif;
    line-height:26px;
    font-size:20px;
    margin:20px 0;
    &::first-letter { 
    font-size: ${(props) => (props.largeFirstLetter ? '50px' : '20px')};
    }
`;
