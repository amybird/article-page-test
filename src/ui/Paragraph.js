import React from 'react';
import styled from 'styled-components/macro';

const Paragraph = ({largeFirstLetter, children}) => (
    <Text largeFirstLetter={largeFirstLetter}>{children}</Text>
)

const Text = styled.p`
    font-family: serif;
    line-height:26px;
    font-size:20px;
    margin:20px 0;
    &::first-letter { 
    font-size: ${props => props.largeFirstLetter ? '50px' : '20px'};
    }
`;

export default Paragraph;