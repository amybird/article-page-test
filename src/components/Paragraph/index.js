import React from 'react';
import PropTypes from 'prop-types';
import { Text } from './styles';

const Paragraph = ({ largeFirstLetter, children }) => (
  <Text largeFirstLetter={largeFirstLetter}>{children}</Text>
);

Paragraph.defaultProps = {
  largeFirstLetter: false
};

Paragraph.propTypes = {
  largeFirstLetter: PropTypes.bool,
  children: PropTypes.string.isRequired
};

export default Paragraph;
