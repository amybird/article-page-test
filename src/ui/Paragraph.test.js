import React from 'react';
import Paragraph from './Container';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

it('renders a paragraph', () => {
  const tree = renderer
    .create(<Paragraph>Testing a paragraph wrapper</Paragraph>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
