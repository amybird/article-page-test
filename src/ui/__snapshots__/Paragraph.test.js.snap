// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a paragraph 1`] = `
.c0 {
  box-sizing: border-box;
  margin-left: auto;
  margin-right: auto;
  width: 100%;
  max-width: 1080px;
}

<div
  className="c0"
>
  Testing a paragraph wrapper
</div>
`;
